python/test/unit/language/test_pipeliner.py

```diff
@@ -302,13 +302,24 @@ def test_pipeline_matmul(scale, device):
             # 2. check sync point
             assert ttgir.count("num = 0 : i32") == 1, "only one sync point for the loads after the loop"
             # 3. check alloc
-            assert ttgir.count("ttg.local_alloc") == (3 if scale else 2), "alloc number not match"
+            if torch.cuda.get_device_capability()[0] == 10:
+                if scale:
+                    # A, B, scale, decomposed A shmem
+                    # MMA pipelining fails to identify the MMA pattern in this case, so the barrier is not inserted.
```

Comment:
Is this alright or is it something we should fix?

Reply:
I think it is alright. Right now we decide to not pipeline because it would require double buffering the TMEM scales and make cause us to run out of TMEM. It might evolve in the future but for now this is probably fine

Reply:
> Right now we decide to not pipeline because it would require double buffering the TMEM scales

I think it is more of a limitation of this code which fails to identify the MMA pattern when there is something between load and MMA. If the perf of emulated scaled dot on Blackwell is important, this should be fixed. 

https://github.com/triton-lang/triton/blob/main/lib/Dialect/TritonGPU/Transforms/Utility.cpp#L1242-L1256